x-pack/legacy/plugins/lens/public/editor_frame_plugin/initializable_component.tsx

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useState, useEffect } from 'react';
+
+interface Props<T> {
+  watch: unknown[];
+  init: () => Promise<T>;
+  render: (props: T) => JSX.Element | null;
+}
+
+export function InitializableComponent<T>(props: Props<T>) {
```

Comment:
Would you expect this component to handle errors by displaying a warning visibly?

Reply:
I changed my mind on this (as seen in the updated PR). I think the error handling can be done entirely by the caller, and InitializableComponent can remain agnostic to it.